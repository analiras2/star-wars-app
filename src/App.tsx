import React, { Component } from "react";
import { Image, StatusBar } from "react-native";
import { colors } from "./res/colors";
import SWButton from "./components/SWButton";
import SWView from "./components/SWView";

interface Props {}
export default class App extends Component<Props> {
  render() {
    return (
      <SWView flex={1} align="center">
        <StatusBar
          animated
          backgroundColor={colors.primary}
          barStyle="light-content"
        />
        <Image
          style={{ position: "absolute", top: 0, bottom: 0, width: "100%" }}
          source={require("./res/img/space.jpg")}
        />
        <Image
          style={{ width: 280, height: 200, marginTop: 38 }}
          source={require("./res/img/logo.png")}
        />
        <SWButton
          top={150}
          color={colors.accent}
          title="Ligar Motores"
          onPress={() => {}}
        />
      </SWView>
    );
  }
}
