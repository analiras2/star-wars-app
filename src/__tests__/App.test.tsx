import { shallow } from "enzyme";
import React from "react";
import { render } from "react-native-testing-library";
import renderer from "react-test-renderer";
import App from "../App";

describe("App", () => {
  const wrapper = shallow<App>(<App />);
  const { getByText } = render(<App />);

  describe("validating App texts", () => {
    it("should render a Ligar Motores", () => {
      expect(getByText(/Ligar Motores/i)).toBeDefined();
    });
  });

  describe("rendering App components", () => {
    it("should render a <SWView />", () => {
      expect(wrapper.find("SWView")).toHaveLength(1);
    });

    it("should render a <SWButton />", () => {
      expect(wrapper.find("SWButton")).toHaveLength(1);
    });

    it("should render two <Image />", () => {
      expect(wrapper.find("Image")).toHaveLength(2);
    });
  });

  it("renders App correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
