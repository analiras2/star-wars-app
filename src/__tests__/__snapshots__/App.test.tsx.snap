// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/res/img/space.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/res/img/logo.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginTop": 38,
        "width": 280,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#edb211",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 150,
        "opacity": 1,
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "auto",
          "color": "#000",
          "flex": 0,
          "fontFamily": "DIN-Bold",
          "fontSize": 18,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "textAlign": "center",
        }
      }
    >
      LIGAR MOTORES
    </Text>
  </View>
</View>
`;
